mm2src/coins/lightning/ln_utils.rs

```diff
@@ -336,8 +490,244 @@ async fn ln_p2p_loop(ctx: MmArc, peer_manager: Arc<PeerManager>, listener: TcpLi
+) {
+    // Retrieve transaction IDs to check the chain for un-confirmations
+    let channel_manager_relevant_txids = channel_manager.get_relevant_txids();
+    let chain_monitor_relevant_txids = chain_monitor.get_relevant_txids();
+
+    for txid in channel_manager_relevant_txids {
+        if filter
+            .platform_coin
+            .as_ref()
+            .rpc_client
+            .get_transaction_bytes(H256::from(txid.as_hash().into_inner()).reversed())
+            .compat()
+            .await
+            .is_err()
+        {
+            // If it's a connection error this will try to broadcast an already successfully broadcasted transaction
+            // which will be rejected, causing no problems and the transaction will be confirmed with transactions_confirmed
+            // later anyways
+            channel_manager.transaction_unconfirmed(&txid);
+        }
+    }
+
+    for txid in chain_monitor_relevant_txids {
+        if filter
+            .platform_coin
+            .as_ref()
+            .rpc_client
+            .get_transaction_bytes(H256::from(txid.as_hash().into_inner()).reversed())
+            .compat()
+            .await
+            .is_err()
+        {
+            chain_monitor.transaction_unconfirmed(&txid);
+        }
+    }
+
+    let mut registered_txs = filter.registered_txs.lock().await;
+    let mut transactions_to_confirm = Vec::new();
+    for (txid, scripts) in registered_txs.clone() {
+        match filter
+            .platform_coin
+            .as_ref()
+            .rpc_client
+            .get_transaction_bytes(H256::from(txid.as_hash().into_inner()).reversed())
+            .compat()
+            .await
+        {
+            Ok(bytes) => {
+                let transaction: Transaction = deserialize(&bytes.into_vec()).expect("Can't deserialize transaction");
+                for (index, vout) in transaction.output.iter().enumerate() {
+                    if scripts.contains(&vout.script_pubkey) {
+                        let script_hash = hex::encode(electrum_script_hash(vout.script_pubkey.as_ref()));
+                        let history = client
+                            .scripthash_get_history(&script_hash)
+                            .compat()
+                            .await
+                            .unwrap_or_default();
+                        for item in history {
+                            if item.tx_hash == H256::from(txid.as_hash().into_inner()).reversed() {
+                                // If a new block mined the transaction while running process_txs_confirmations it will be confirmed later in ln_best_block_update_loop
+                                if item.height > 0 && item.height <= current_height as i64 {
+                                    let header = match client
+                                        .blockchain_block_header(
+                                            item.height.try_into().expect("Convertion to u64 should not fail"),
```

Comment:
As I understand the negative height gets returned if the transaction is unconfirmed which is what's needed here. I also used `blockchain.scripthash.get_history` because this is the only electrum method that returns the height. Please correct me If I am wrong.

Reply:
We never know what kind of trash some external service may return :D